### PR TITLE
build: allow Envoy mobile to build sans YAML.

### PR DESCRIPTION
**Pull request by alyssawilk**
e2e tests will not pass until https://github.com/envoyproxy/envoy/pull/26451 lands

Risk Level: medium (should be no-op but includes refactors)
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
part of https://github.com/envoyproxy/envoy/issues/26450

### CODE REVIEW

File: .github/workflows/mobile-compile_time_options.yml [hunk 1]
```diff
@@ -110,6 +110,6 @@ jobs:
           --define=envoy_mobile_request_compression=disabled \
           --define=envoy_enable_http_datagrams=disabled \
           --define=google_grpc=disabled \
-          --@envoy//bazel:http3=False \
```
Comment (RyanTheOptimist) [inline .github/workflows/mobile-compile_time_options.yml:113]: Is this removal a mistake, or intentional?

**Review by RyanTheOptimist — APPROVED** [review]
Wow, this is kinda amazing.LGTM though I do have one question

**Comment by htuch** [comment]
@alyssawilk do you want to merge after answering @RyanTheOptimist ? thanks.

File: .github/workflows/mobile-compile_time_options.yml (hunk 1 above)
Comment (alyssawilk) [reply .github/workflows/mobile-compile_time_options.yml:113]: intentional - I wanted to make sure H3 builds worked for !yaml for our internal use cases

**Comment by moderation** [comment]
What happens if I try this on a server build? I recall another comment I can't find saying there was an /admin dependency on YAML

**Comment by alyssawilk** [comment]
likely it simply won't build.  For sure until the runtime PR lands things won't work well.  I'm only trying to get yamlless for E-M due to size constraints but I have no objection to someone else extending the project to more of Envoy!

**Pull request merged**